include/rcpputils/filesystem_helper.hpp

```diff
@@ -137,6 +137,17 @@ class path
     return split_fname.size() == 1 ? path("") : path("." + split_fname.back());
   }
 
+  path remove_extension(int n_times = 1)
+  {
+    for (int i = 0; i < n_times; i++) {
+      size_t last_dot = path_.find_last_of('.');
+      if (last_dot == std::string::npos) {
+        return path(path_);
+      }
+      return path(path_.erase(last_dot, path_.size() - 1));
```

Comment:
```suggestion
      temp_path = path_.erase(last_dot, path_.size() - 1);
      }  // end for-loop 
      
      return temp_path;
```

Reply:
We've agreed on the following 
```
 path_.erase(last_dot, path_.size() - 1)
 return *this;
```
since we will be modifying the actual `path_` string.

Reply:
Sounds good. As long as the second `return` is outside the for-loop.